src/module_acquisition.md

```diff
@@ -4,21 +4,17 @@ Different languages have had differing dependency resolution mechanisms.
-
-Specifying the version of the code you want is done by specifying the tag or SHA you wish to use. If you can support multiple versions (particularly if you're writing a library instead of acting as an end user) you can also provide an array of tags/shas or a special Semver object defining the major.minor.patch ranges acceptable. (This will depend on getting compound types in the type system first)
-
-The dependencies will be added to the `dependencies` directory, which will also be in the `.gitignore` list created by `alan init`. These dependencies, once loaded, will check their own needed dependencies. If there is a dependency "above" them that contains a `git` repo with a valid tag, it's done. If it finds a `git` repo with an invalid tag, it will install its needed version locally in its own `dependencies` sub-directory. If it finds no match at all in the parent hierarchy, it will place its dependency in the `dependencies` directory it finds itself in to potentially reduce duplication of commonly-used third party dependencies.
-
-At least that'll be the first pass -- lots of potential optimizations can be taken from "yarn": determine if there are conflicting dependency needs across the entire tree and raise the most common version up while leaving the less common versions to be nested, instead of whichever was the first encountered. Also compute diffs of the dependency hierarchy before and after a dependency is added or removed and only perform the necessary operations instead of redoing everything.
+Long-term, though, `alan` intends to have a centralized package hosting system because there are several advantages to the community for such a thing:
 
+1. The packages can be guaranteed immutable to make sure production deployments are more reliable.
+2. The central repository can require package signing to make it harder for popular packages to be hijacked and have malware injected into them (and also raise the bar to make sure only more serious projects are submitted).
+3. The semantic versioning (major, minor, patch) meaning can be better trusted by requiring published packages to conform to a validation mechanism we call "Semantic BDD." (TODO: Write up blog post on Semantic BDD, and link to it here.)
+4. Finally, the central repository can provide discoverability and recommendation mechanisms to developers on which packages may be worth using versus others. Simple metrics like download and star counts are one, but beyond that there could be recommendations based on things starred by other developers that also starred things you have, or the total coverage of their Semantic BDD tests versus the entirety of their export scope surface area.
```

Comment:
nit: let's not number these? It makes me think they are prioritized and I personally think people would care most about 4

Reply:
Sure, I can turn them into bullet points. However I personally view that order as the order I care about even as an end user. :)

A recommendation engine for packages is something I don't give a crap about. I'll install what I want to install, and having guarantees that it won't disappear on me and is reasonably sure to be by the actual author and following semantic versioning are more important.